src/univariate/continuous/normal.jl

```diff
@@ -27,21 +27,25 @@ External links
 * [Normal distribution on Wikipedia](http://en.wikipedia.org/wiki/Normal_distribution)
 
 """
-struct Normal{T<:Real} <: ContinuousUnivariateDistribution
+struct Normal{T<:Number} <: UnivariateDistribution{ContinuousSupport{T}}
     μ::T
     σ::T
-    Normal{T}(µ::T, σ::T) where {T<:Real} = new{T}(µ, σ)
+    Normal{T}(µ::T, σ::T) where {T<:Number} = new{T}(µ, σ)
 end
 
-function Normal(μ::T, σ::T; check_args=true) where {T <: Real}
-    check_args && @check_args(Normal, σ >= zero(σ))
+function Normal(μ::T, σ::T; check_args=true) where {T <: Number}
+    if check_args
+        @check_uncountable(Normal, μ)
+        @check_args(Normal, σ >= zero(σ))
+    end
     return Normal{T}(μ, σ)
 end
 
 #### Outer constructors
-Normal(μ::Real, σ::Real) = Normal(promote(μ, σ)...)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Rational, σ::Rational) = Normal(float(μ), float(σ))
```

Comment:
Ah, we have a constraint: the mean and the eltype of the distribution need to agree.  Makes sense for a `Normal`, it can neither be sampled nor the pdf evaluated in `Rational`.